### PR TITLE
[WIP] Change deployment's API version to apps/v1beta2 

### PR DESCRIPTION
**Pull request by crimsonfaith91**
**What this PR does / why we need it**:
This PR changes deployment controller to use `apps/v1beta2` APIs.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: xref #51049

**Special notes for your reviewer**:
(1) We have to make a copy of `extensions/v1beta1` deployment util files as `kubectl` needs to support +/- 2 versions. I am looking for a way to de-duplicate most of the codes.
(2) The integration and e2e tests will still use `extensions/v1beta1` APIs.

**Release note**:

```release-note
Change deployment's API version to apps/v1beta2.
```

### CODE REVIEW

File: pkg/controller/deploymentv1beta1/deployment_util.go [hunk 1]
```diff
@@ -0,0 +1,1006 @@
+/*
```
Comment (sttts) [inline pkg/controller/deploymentv1beta1/deployment_util.go]: these are really really really ugly. Tons of duplicated code. Moreover, this has nothing to do with a controller anymore and hence should not be under pkg/controller.

File: pkg/controller/deploymentv1beta1/deployment_util.go [hunk 2]
```diff
@@ -0,0 +1,1006 @@
...
+	NewRSAvailableReason = "NewReplicaSetAvailable"
+	// TimedOutReason is added in a deployment when its newest replica set fails to show any progress
+	// within the given deadline (progressDeadlineSeconds).
+	TimedOutReason = "ProgressDeadlineExceeded"
+	// PausedDeployReason is added in a deployment when it is paused. Lack of progress shouldn't be
+	// estimated once a deployment is paused.
+	PausedDeployReason = "DeploymentPaused"
+	// ResumedDeployReason is added in a deployment when it is resumed. Useful for not failing accidentally
+	// deployments that paused amidst a rollout and are bounded by a deadline.
+	ResumedDeployReason = "DeploymentResumed"
```
Comment (sttts) [inline pkg/controller/deploymentv1beta1/deployment_util.go]: What are all these constants? Aren't they part of the external types.go? If not, why not?

File: pkg/controller/deploymentv1beta1/deployment_util.go (hunk 1 above)
Comment (sttts) [reply pkg/controller/deploymentv1beta1/deployment_util.go]: With kubectl moving out of main kube repo, what is the plan with these? @fabianofranz @pwittrock

File: pkg/controller/deploymentv1beta1/deployment_util.go (hunk 2 above)
Comment (kargakis) [reply pkg/controller/deploymentv1beta1/deployment_util.go]: Yes, this is api and should be moved under types.go.

File: pkg/controller/deploymentv1beta1/deployment_util.go (hunk 1 above)
Comment (crimsonfaith91) [reply pkg/controller/deploymentv1beta1/deployment_util.go]: @sttts Thanks for the feedback. This PR is WIP and not intended for review without getting rid of the duplicated codes. I am still finding a way to de-duplicate the util codes. I will mark the PR as WIP, and inform you when it is ready for review.

File: pkg/controller/deploymentv1beta1/deployment_util.go (hunk 2 above)
Comment (crimsonfaith91) [reply pkg/controller/deploymentv1beta1/deployment_util.go]: Noted - thanks!

File: pkg/controller/deploymentv1beta1/deployment_util.go (hunk 1 above)
Comment (sttts) [reply pkg/controller/deploymentv1beta1/deployment_util.go]: oh, ok. Just saw it being assigned to me :)

Comment (crimsonfaith91) [reply pkg/controller/deploymentv1beta1/deployment_util.go]: Sorry for not marking the PR WIP beforehand. Thanks for the feedback! :)

**Comment by kargakis** [comment]
Will this be pursued for 1.8?

**Comment by crimsonfaith91** [comment]
@kargakis No. It is meant for 1.9 :)

**Comment by kargakis** [comment]
@crimsonfaith91 needs a rebase

**Comment by dims** [comment]
@crimsonfaith91 @sttts Do we move this out of 1.9 milestone?

**Comment by crimsonfaith91** [comment]
@dims I believe so, but please let me confirm with others. I will get back to you tomorrow. Sorry for the inconvenience.

**Comment by crimsonfaith91** [comment]
@dims This (comment)[https://github.com/kubernetes/kubernetes/issues/51049#issuecomment-345080702] answers your question.

**Pull request closed without merging**